### ISSUE TITLE
Freshmark doesn't work on JDK 15 due to missing Nashorn

### ISSUE THREAD

**Issue by LunNova** [diffplug/spotless#803]
Nashorn was removed in JDK 15, and freshmark [depends on it](https://github.com/diffplug/jscriptbox/blob/master/src/main/java/com/diffplug/jscriptbox/javascript/Nashorn.java#L42) so crashes.

```
Caused by: java.lang.RuntimeException: Error on line 21: Cannot invoke "javax.script.ScriptEngine.getContext()" because "jsEngine" is null
        at com.diffplug.freshmark.Parser$1ErrorFormatter.lambda$wrap$2(Parser.java:89)
        at com.diffplug.freshmark.ParserIntronExon.bodyAndTags(ParserIntronExon.java:71)
        at com.diffplug.freshmark.Parser.compile(Parser.java:162)
        at com.diffplug.freshmark.CommentScript.compile(CommentScript.java:85)
        ... 141 more
Caused by: java.lang.NullPointerException: Cannot invoke "javax.script.ScriptEngine.getContext()" because "jsEngine" is null
        at com.diffplug.jscriptbox.javascript.Nashorn.lambda$language$0(Nashorn.java:43)
        at com.diffplug.jscriptbox.JScriptBox.build(JScriptBox.java:89)
        at com.diffplug.freshmark.FreshMark.setupScriptEngine(FreshMark.java:65)
        at com.diffplug.freshmark.CommentScript.lambda$compileSection$0(CommentScript.java:71)
        at com.diffplug.common.base.Errors$Rethrowing.lambda$wrap$11(Errors.java:342)
        at com.diffplug.common.base.Errors$Rethrowing.get(Errors.java:334)
        at com.diffplug.freshmark.CommentScript.compileSection(CommentScript.java:70)
        at com.diffplug.freshmark.Parser$1State.tag(Parser.java:144)
        at com.diffplug.freshmark.Parser$1ErrorFormatter.lambda$wrap$2(Parser.java:78)
        ... 144 more

```

Gradle version: 6.8.2
JDK version: AdoptOpenJDK jdk-15.0.1.9-hotspot
OS version: Windows 10 20H2
Freshmark config:

```java
spotless.freshmark(it -> {
	it.properties(props -> props.putAll(settings.toProperties(project)));
	it.target(files(project, "**/*.md"));
	it.indentWithTabs();
	it.endWithNewline();
});
```

This can be worked around by adding nashorn-core to the buildscript classpath:

```groovy
buildscript {
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.openjdk.nashorn:nashorn-core:15.2")
	}
}
```

**Comment by nedtwigg** [comment]
Thanks! We should build this into Spotless transparently. We can do it like this:

- Move [`JreVersion`](https://github.com/diffplug/spotless/blob/main/testlib/src/main/java/com/diffplug/spotless/JreVersion.java) from `testlib` into `lib`, keep the `assume()` methods in testlib in a new class `JreVersionAssume`
- Refactor [`GoogleJavaFormatStep`](https://github.com/diffplug/spotless/blob/9dd69f7ff7feb6b2b7d26285a5b5ea6896a935ad/lib/src/main/java/com/diffplug/spotless/java/GoogleJavaFormatStep.java#L79-L100) to use that too
- Depending on `JreVersion`, add the `nashorn-core` dep into the `JarState` below

https://github.com/diffplug/spotless/blob/9dd69f7ff7feb6b2b7d26285a5b5ea6896a935ad/lib/src/main/java/com/diffplug/spotless/markdown/FreshMarkStep.java#L57

I'm still on Java 8 personally, so this isn't going to make the top of my TODO anytime soon. Happy to take a PR! We're [already doing CI on JRE 15](https://github.com/diffplug/spotless/blob/9dd69f7ff7feb6b2b7d26285a5b5ea6896a935ad/.circleci/config.yml#L75-L80), so the testing part should be pretty straightforward.

**Comment by nedtwigg** [comment]
Released in `plugin-gradle 6.14.1`.
